sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceId;
+import java.util.Random;
+
+final class RandomIdsGenerator implements IdsGenerator {
+  private static final long INVALID_ID = 0;
+  // TODO change with ThreadLocal version
+  // https://github.com/open-telemetry/opentelemetry-java/issues/406
+  private final Random random;
+
+  RandomIdsGenerator(Random random) {
+    this.random = random;
+  }
+
+  @Override
+  public SpanId generateSpanId() {
+    long id;
+    do {
+      id = random.nextLong();
+    } while (id == INVALID_ID);
+    return new SpanId(id);
+  }
+
+  @Override
+  public TraceId generateTraceId() {
+    long idHi;
+    long idLo;
+    do {
```

Comment:
Recommend encapsulating the `do { ... } while (id is invalid)` logic into a method and leveraging it from both generate methods.

Reply:
https://github.com/open-telemetry/opentelemetry-java/issues/642